File: 11/binary_search.rb
def binary_search(array, number)
  index = array.size / 2
  median = array[index]

  if array.size == 1
    if array.first == number
      return number
    else
      return
    end
  end

  if number > median
    binary_search array[index + 1..-1], number
  elsif number < median
    binary_search array[0...index], number
  else
    median
  end
end

File: 11/python/binary_search.py
def binary_search(element, array):
    array_length = len(array)
    middle_index = array_length / 2
    median = array[middle_index]

    if element == median:
        return median
    elif array_length <= 1:
        return None
    elif element > median:
        return binary_search(element, array[middle_index + 1:])
    else:
        return binary_search(element, array[:middle_index])

def test():
    array = range(1, 10)
    if binary_search(5, array) == 5 and binary_search(100, array) == None:
        print "Binary search test successful"
    else:
        print "Binary search test failed"

test()
